### PR TITLE
[MIG][14.0] base_view_inheritance_extension

### PR DESCRIPTION
**Pull request by alexis-via**
I don't really understand the code, so please check this PR carefully.

### CODE REVIEW

**Comment by youring** [comment]
Hello,  here is what I tried, create product from sale order line, defaults to Storable Product:
```
    <record id="view_order_form" model="ir.ui.view">
        <field name="model">sale.order</field>
        <field name="inherit_id" ref="sale.view_order_form" />
        <field name="arch" type="xml">
            <xpath
                expr="//field[@name='order_line']/tree//field[@name='product_id']"
                position="attributes">
                <attribute
                    name="context"
                    operation="python_dict"
                    key="default_type"
                >product</attribute>
            </xpath>
        </field>
    </record>
````
But this produces error when install my custom module: 
```odoo.tools.convert.ParseError: "Error while validating view
Field 'product' used in attributes must be present in view but is missing:
```

Then I tried to change key value product to "product",  with quote,  it installs without error. But produces this error when creating product from sale order line:

```ValueError: Wrong value for product.product.type: '"product"'```

I wonder if adding/changing default value for selection fields in context is supported by this module? Or how to change the context properly?
Thank you.

**Comment by alexis-via** [comment]
@youring change product to product_id !!! The error message is correct !

**Comment by youring** [comment]
@alexis-via Maybe you misunderstood my points. "product" is one of the selection of the type field, what if I want to create a service type product by default when creating from sale order line?

**Comment by alexis-via** [comment]
ok, I understand now why you need the quotes... I must say I never used base_view_inheritance that way, so I don't know... maybe other people now ?

**Comment by remihb** [comment]
Try quoting 'product', means single quotes '

```
                <attribute
                    name="context"
                    operation="python_dict"
                    key="default_type"
                >'product'</attribute>
```

**Comment by youring** [comment]
@remihb Thanks. I tried but similar error
`ValueError: Wrong value for product.product.type: "'product'"`
Maybe @pedrobaeza and @hbrunn can give some hint. Thanks.

**Comment by pedrobaeza** [comment]
I can't say about such case, sorry.

**Comment by remihb** [comment]
Ok wrong info sry. Similar error for me too.

**Comment by nilshamerlinck** [comment]
Hi @alexis-via 

Could you please include https://github.com/OCA/server-tools/pull/1946 and the `setup/base_view_inheritance_extension/` folder in your PR?

This will be useful for travis builds of tierce repos when this module is needed.

Thank you

**Comment by alexis-via** [comment]
done.

**Comment by phuctranfxvn** [comment]
Hi @alexis-via
Can you add PyYAML as the python dependency to make travis-ci happy ?

**Review by gurneyalex — DISMISSED** [review]

File: base_view_inheritance_extension/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+{
+    "name": "Extended view inheritance",
+    "version": "14.0.1.0.0",
+    "author": "Therp BV,Odoo Community Association (OCA)",
+    "license": "LGPL-3",
+    "category": "Hidden/Dependency",
+    "summary": "Adds more operators for view inheritance",
+    "website": "https://github.com/OCA/server-tools",
+    "depends": ["base"],
```
Comment (gurneyalex) [inline base_view_inheritance_extension/__manifest__.py:12]: from #1946 
```suggestion
    "depends": ["base"],
    "external_dependencies": {"python": ["pyyaml"]},
```

**Comment by jcoux** [comment]
@alexis-via Just created a PR on your branch to fix a bug:
https://github.com/akretion/server-tools/pull/28

File: base_view_inheritance_extension/__manifest__.py (hunk 1 above)
Comment (yvaucher) [reply base_view_inheritance_extension/__manifest__.py:12]: @alexis-via this is marked as resolved but the travis error is still there. Was it lost during a rebase?

(marking it as unsolved)

**Comment by simahawk** [comment]
```
2021-05-17 07:17:04,278 6179 INFO openerp_test odoo.modules.loading: Loading module base_view_inheritance_extension (11/28)

2021-05-17 07:17:04,316 6179 CRITICAL openerp_test odoo.modules.module: Couldn't load module base_view_inheritance_extension

2021-05-17 07:17:04,317 6179 CRITICAL openerp_test odoo.modules.module: No module named 'yaml'

2021-05-17 07:17:04,319 6179 WARNING openerp_test odoo.modules.loading: Transient module states were reset

2021-05-17 07:17:04,319 6179 ERROR openerp_test odoo.modules.registry: Failed to load registry

2021-05-17 07:17:04,320 6179 CRITICAL openerp_test odoo.service.server: Failed to initialize database `openerp_test`.

```
can we activate MQT=PIP here? :pray: 

If not, at least enable automatic generation of requirements via pre-commit :wink:

**Comment by simahawk** [comment]
@alexis-via seems you've removed https://github.com/akretion/server-tools/pull/28 in your last push force, correct?

**Review by yvaucher — APPROVED** [review]

**Review by simahawk — APPROVED** [review]

**Pull request merged**